lib/mayaUsd/nodes/stageData.h

```diff
@@ -75,39 +75,9 @@ class MayaUsdStageData : public MPxGeometryData
      */
     //@{
 
-    // The original Pixar code was the following:
-    //
-    // UsdStageRefPtr stage;
-    //
-    // Now using a weak pointer instead of a referencing pointer.  The AL
-    // plugin originally used a referencing pointer, but ran into problems.
-    // From a Rob Bateman @ AL e-mail, 7-Apr-2019:
-    //
-    // "The reason for the weak pointer was that we found that Maya seemed
-    // to have a memory leak (as in, the MPxData derived objects were being
-    // created, but never deleted when we expected them to be - possibly by
-    // design?). We had originally used a direct mirror of the Pixar data
-    // object, but because the data objects were never freed (even after
-    // file new), the original stage would be retained. This meant the
-    // internal SdfLayerCache in USD would keep hold of the previously
-    // loaded layers. So we found we had this problem:
-    //
-    // 1. Import some.usda file into a proxy shape.
-    // 2. Make some modifications
-    // 3. File New
-    // 4. Import the same "some.usda" file into a proxy shape.
-    //
-    // At this point, USD would essentially hand you back the stage
-    // composed of the modified layers, rather than a clean stage composed
-    // from the files on disk. Switching the type from a shared_ptr to a
-    // weak_ptr worked around this issue. Basically if we use a shared
-    // pointer here, the only way to reload a scene, would be to restart
-    // Maya."
-    //
-    // Logged as https://github.com/Autodesk/maya-usd/issues/528
-
-    UsdStageWeakPtr stage;
-    SdfPath         primPath;
+    UsdStageRefPtr stage;
```

Comment:
Thanks to @kxl-adsk  for finding this out.

In the new code in copyInternalData(), we could be dealing with a file backed stage where we do not transfer the content and rather point to the same stage.  Having a weak pointer to the stage in the data block would be problematic.

Reply:
If this is no longer a problem, then we should re-test
https://github.com/Autodesk/maya-usd/issues/528
and close it, if it's no longer reproducible after your fix.

Reply:
Good point. Looking into it.

Reply:
@ppt-adsk addressed in 4b8947b158e47d0e2f00802d49e56e3faf9d3fa2

We no longer need the workaround.

Reply:
@HamedSabri-adsk @ppt-adsk I haven't yet reviewed the code, but this is change is outside of the scope for the duplicate implementation. Let's take it out of this PR and open a new one after.

Reply:
Ok, I added the change here since it was relevant. I will revert the change and open a separate PR if it makes thing easier.

Reply:
Thank you, Hamed.

Reply:
FYI. @pmolodo @mattyjams @fabal